rfcs/0002-validation/text.md

```diff
@@ -0,0 +1,81 @@
+dataset.
+
+## Problem Statement | Background
+
+There isn't a broad community consensus around how single cell datasets should be stored and annotated. This
+makes it harder to integrate these datasets, which is bad because that's a main aim of our single cell work. So, we have
+a simple schema that we want all datasets to adhere to. If all the datasets we store at least follow this schema, then
+our data corpus will be usable for data integration research and the production of atlases.
+
+The problem is that getting a dataset to follow the schema isn't trivial. The schema is pretty lightweight, but there's
+a bunch of data munging, relabeling of columns, etc. that's at best a pain and at worst beyond the capabilities of some
+submitters. So we want some tooling to help with that, and we also want to make sure that before we actually add a
+dataset to the portal/cellxgene we verify that it follows the schema.
+
+Finally, single cell datasets come in different formats that are associated with different toolchains. We want to
+support as many of those toolchains as we can, so we're going to convert datasets into different formats. It seems to
+make sense to do this right after validation is complete since we're already parsing the original file anyway, and
+there's not really a good reason to delay converting.
+
+
+## Product Requirements
+
+The key requirement here is that users can prepare their own datasets without intervention from CZI. This goes along
+with all the other work to support self-submission: file upload, collection create, private/public publication.
+
+
+## Detailed Design | Architecture | Implementation
+
+There are two main phases for the preparation of a dataset. In the first, the submitter is working with their dataset
+locally, preparing it so it follows the schema and is in an accepted file format. We have command line tools to help
+with that are part of cellxgene. The second phase occurs after the dataset file is uploaded, and that it the phase we
+are mainly concerned about here.
+
+Through some upload process (described in an RFC elsewhere) a dataset file ends up in S3. Initially, that file is
+expected be in the AnnData HDF5 format, but eventually other formats will be supported as well. Once that happens, we
+need to validate and convert that file, which occurs in a number of steps:
+
+1. Localize the uploaded file to the local storage of some instance (HDF5 doesn't like remote files usually).
+2. Run the validation tool on the file. This is the same tool that's available to users via the cellxgene CLI, so this
+   is really just verifying that they ran it. This should only take a couple of seconds and use little memory.
+3. Update the database with the result of validation. If it fails, stop here.
+4. Convert the uploaded file to the Loom format.
+5. Convert the uploaded file to a Seurat object and save it.
+6. Convert the uploaded file to a SingleCellExperiment object and save it.
```

Comment:
Is this an emerging (undocumented) requirement? Previously, I've also seen references to three formats .

Reply:
No not really, I'll remove it. It's the natural next format, but we should go with what's in the docs.